### PR TITLE
fix: cm secret patch dry run

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Pages/Shared/ConfigMapSecret/utils.ts [hunk 1]
```diff
@@ -265,7 +266,12 @@ export const getConfigMapSecretFormInitialValues = ({
             selectedType: type ?? configMapSecretMountDataMap.environment.value,
```
Comment (AbhishekA1509) [inline src/Pages/Shared/ConfigMapSecret/utils.ts:266]: Can we create a base struct that is common in both returns?

File: src/Pages/Shared/ConfigMapSecret/utils.ts [hunk 2]
```diff
@@ -642,6 +650,7 @@ export const getConfigMapSecretDraftAndPublishedData = ({
         ) {
             draftData = {
                 ...draftConfigData,
+                parsedData: JSON.parse(draftConfigData.data),
```
Comment (AbhishekA1509) [inline src/Pages/Shared/ConfigMapSecret/utils.ts:653]: Do we need try catch?

Comment (RohitRaj011) [reply src/Pages/Shared/ConfigMapSecret/utils.ts:653]: no this data will always be present

File: src/config/utils.ts [hunk 3]
```diff
@@ -27,3 +32,16 @@ export const checkIfDevtronOperatorHelmRelease = (
...
+
+/**
+ * Checks if the provided pathname matches the current path.
+ * If the paths do not match, returns a custom message or a default unsaved changes prompt.
+ *
+ * @param currentPathName - The current path to compare against.
+ * @param customMessage - Optional custom message to display when the path does not match.
+ * @returns A function that takes an object with a `pathname` property and performs the path match check.
+ */
+export const checkIfPathIsMatching =
```
Comment (AbhishekA1509) [inline src/config/utils.ts:44]: Can we set the return type from PromptProps, and if we are keeping it in dashboard ideally it should be in Helpers.tsx

File: src/Pages/Shared/ConfigMapSecret/utils.ts (hunk 1 above)
Comment (AbhishekA1509) [reply src/Pages/Shared/ConfigMapSecret/utils.ts:270]: We can destructure configMapSecretData above the if statement and re-apply same conditions like externalType, filePermission, roleARN, isResolvedData, yamlMode, isFilePermissionChecked and even more

File: src/Pages/Shared/ConfigMapSecret/ConfigMapSecret.service.ts [hunk 4]
```diff
@@ -224,7 +228,7 @@ export const getConfigMapSecretManifest = async (params: ConfigMapSecretManifest
                         ? ConfigResourceType.ConfigMap
                         : ConfigResourceType.Secret,
             },
-            { signal },
+            { signal: abortControllerRef?.current?.signal },
```
Comment (AbhishekA1509) [inline src/Pages/Shared/ConfigMapSecret/ConfigMapSecret.service.ts:231]: Can we send ref instead?

File: src/Pages/Shared/ConfigMapSecret/ConfigMapSecretContainer.tsx [hunk 5]
```diff
@@ -529,13 +579,46 @@ export const ConfigMapSecretContainer = ({
     }
 
     const handleNoOverrideFormCancel = () => {
-        setDefaultMergeStrategy()
-        setFormState({ type: 'RESET' })
+        // setFormState({ type: 'RESET' })
```
Comment (AbhishekA1509) [inline src/Pages/Shared/ConfigMapSecret/ConfigMapSecretContainer.tsx:582]: Can remove

**Review by AbhishekA1509 — APPROVED** [review]
Approving on pretext that comments would be resolved in single branch

**Pull request merged**